### PR TITLE
AVRO-2424: Make Py2 bindings work on setuptools' development mode

### PR DESCRIPTION
**Pull request by sekikn**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO/) issues and references them in the PR title. For example, "AVRO-1234: My Avro PR"
  - https://issues.apache.org/jira/browse/AVRO-2424
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

This change is for the build process and difficult to write unit tests. Instead, I ran `python setup.py develop`, `pip install -e .`, and `./build.sh dist` manually and confirmed that they worked as expected.

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: lang/py/setup.py [hunk 1]
```diff
@@ -25,11 +27,19 @@
 if version_info[:2] <= (2, 5):
     install_requires.append('simplejson >= 2.0.9')
 
+curdir = os.getcwd()
+version_file = ("VERSION.txt" if os.path.isfile("VERSION.txt")
```
Comment (Fokko) [inline lang/py/setup.py]: Can we wrap this in a function? This would allow us to write a test as well :-)

Comment (sekikn) [reply lang/py/setup.py]: Thanks for the review @Fokko! I'll update the PR.

File: lang/py/.gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
+avro.egg-info/
```
Comment (kojiromike) [inline lang/py/.gitignore]: Could we instead put a generic `*.egg-info/` in the root .gitignore?

File: pom.xml [hunk 3]
```diff
@@ -289,6 +289,7 @@
                 <exclude>lang/perl/pm_to_blib</exclude>
                 <exclude>lang/perl/blib/**/.exists</exclude>
                 <exclude>lang/py/build/**</exclude>
+                <exclude>lang/py/src/avro.egg-info/**</exclude>
```
Comment (kojiromike) [inline pom.xml]: Same as my comment to .gitignore

File: lang/py/setup.py [hunk 4]
```diff
@@ -23,13 +25,22 @@
...
+  install_requires.append('simplejson >= 2.0.9')
+
+def _get_version():
+  curdir = os.getcwd()
+  version_file = ("VERSION.txt" if os.path.isfile("VERSION.txt")
+    else os.path.join(curdir[:curdir.index("lang/py")], "share/VERSION.txt"))
+  with open(version_file) as verfile:
+    # To follow the naming convention defined by PEP 440
+    # in the case that the version is like "x.y.z-SNAPSHOT"
+    return verfile.read().rstrip().replace("-", "+")
```
Comment (kojiromike) [inline lang/py/setup.py]: Nice! I'm a fan.

**Comment by sekikn** [comment]
Thanks for the review @kojiromike! Just updated the PR.

File: lang/py/setup.py (hunk 1 above)
Comment (Fokko) [reply lang/py/setup.py]: Thanks!

**Comment by sekikn** [comment]
Resolved conflicts.

**Pull request merged**